builder/azure/common/template/template_builder.go

```diff
@@ -450,6 +450,22 @@ func (s *TemplateBuilder) SetBootDiagnostics(diagSTG string) error {
 	return nil
 }
 
+func (s *TemplateBuilder) SetSecurityProfile(secureBootEnabled bool, vtpmEnabled bool) error {
+	s.setVariable("apiVersion", "2020-12-01") // Required for Trusted Launch
+	resource, err := s.getResourceByType(resourceVirtualMachine)
+	if err != nil {
+		return err
+	}
+
+	resource.Properties.SecurityProfile = &compute.SecurityProfile{}
+	resource.Properties.SecurityProfile.UefiSettings = &compute.UefiSettings{}
```

Comment:
Hey @maxilampert this looks good to me but can you add a test to `template_builder_test.go` to capture these new fields, after that happy to merge this and get this functionality in, it's worth calling out that the vTPM info can't be captured in a deployed image, but for your use case for VMs it'll work